### PR TITLE
libphoenix: fix mutexing in pthread_create

### PR DESCRIPTION
**Pull request by adamdebek**
RTOS-651

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by adamdebek** [comment]
https://github.com/phoenix-rtos/phoenix-rtos-project/issues/891

**Review by damianloew — COMMENTED** [review]
When the commit solves some issue we should add the following line in the commit description:
Screenshot from 2023-10-18 12-06-48

**Comment by nalajcie** [comment]
not doing code review, but please use `locking` instead of `mutexing` in commit message title :)

File: pthread/pthread.c [hunk 1]
```diff
@@ -260,6 +260,8 @@ int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
 	ctx->cancelstate = PTHREAD_CANCEL_ENABLE;
 	*thread = (pthread_t)ctx;
 
+	mutexLock(pthread_common.pthread_list_lock);
```
Comment (lukileczo) [inline pthread/pthread.c:263]: Deadlock can happen now - please use `_pthread_ctx_put` if error is returned from `beginthreadex`

Comment (lukileczo) [reply pthread/pthread.c:263]: Note that `_pthread_ctx_put` unlocks the mutex

**Comment by badochov** [comment]
Does it fix zedbaord target as well?

File: pthread/pthread.c [hunk 2]
```diff
@@ -260,23 +260,25 @@ int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
...
+		if (pthread_common.pthread_list != NULL && pthread_common.pthread_list->id == 0) {
 			pthread_common.pthread_list = NULL;
-
+		}
 		LIST_ADD(&pthread_common.pthread_list, ctx);
-		mutexUnlock(pthread_common.pthread_list_lock);
 	}
 
+	mutexUnlock(pthread_common.pthread_list_lock);
+
```
Comment (lukileczo) [inline pthread/pthread.c:281]: Now we may unlock the mutex twice :)
I'd do
```
else {
    if (ctx->detached == PTHREAD_CREATE_JOINABLE) {
        ...
    }
    mutexUnlock()
}
```

**Comment by adamdebek** [comment]
> Does it fix zedbaord target as well?

I didn't see it fail on this target, maybe scheduling differs, but if it was failing on zedboard tihs pr should also fix it.

**Pull request merged**